File: AppbarSample/app/src/main/java/com/example/appbarsample/MainActivity.java
package com.example.appbarsample;

import android.os.Bundle;
import android.support.design.widget.CollapsingToolbarLayout;
import android.support.design.widget.TabLayout;
import android.support.v4.view.ViewPager;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.AppCompatImageView;
import android.view.LayoutInflater;
import android.view.animation.AccelerateInterpolator;
import android.view.animation.AlphaAnimation;
import android.view.animation.Animation;
import android.widget.ImageView;

public class MainActivity extends AppCompatActivity {
    private AppCompatImageView imageView;
    private TabLayout tabLayout;
    private ViewPager viewPager;


    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        findViewById();
        initViewPager();
        initTabView();
    }

    private void findViewById() {
        imageView = findViewById(R.id.img);
        tabLayout = findViewById(R.id.tab);
        viewPager = findViewById(R.id.view_pager);
    }

    private void initViewPager() {
        viewPager.setAdapter(new ChildViewAdapter());
    }

    private void initTabView() {
        tabLayout.setupWithViewPager(viewPager);
        tabLayout.addOnTabSelectedListener(new TabLayout.OnTabSelectedListener() {
            @Override
            public void onTabSelected(TabLayout.Tab tab) {
                fadeInAndShowImage(imageView);
            }

            @Override
            public void onTabUnselected(TabLayout.Tab tab) {

            }

            @Override
            public void onTabReselected(TabLayout.Tab tab) {

            }
        });

        for (int i = 0; i < tabLayout.getTabCount(); i++) {
            TabLayout.Tab tab = tabLayout.getTabAt(i);
            if (tab != null) {
                tab.setCustomView(LayoutInflater.from(this).inflate(R.layout.layout_tab, null, false));
            }
        }
    }

    private int[] imgRes = {R.drawable.google, R.drawable.apple};

    private void fadeOutAndHideImage(final ImageView img) {
        Animation fadeOut = new AlphaAnimation(1, 0);
        fadeOut.setInterpolator(new AccelerateInterpolator());
        fadeOut.setDuration(500);

        fadeOut.setAnimationListener(new Animation.AnimationListener() {
            public void onAnimationEnd(Animation animation) {
            }

            public void onAnimationRepeat(Animation animation) {
            }

            public void onAnimationStart(Animation animation) {
                img.setImageResource(imgRes[tabLayout.getSelectedTabPosition()]);
            }
        });

        img.startAnimation(fadeOut);
    }

    private void fadeInAndShowImage(final ImageView img) {
        Animation fadeIn = new AlphaAnimation(0, 1);
        fadeIn.setInterpolator(new AccelerateInterpolator());
        fadeIn.setDuration(500);

        fadeIn.setAnimationListener(new Animation.AnimationListener() {
            public void onAnimationEnd(Animation animation) {
            }

            public void onAnimationRepeat(Animation animation) {
            }

            public void onAnimationStart(Animation animation) {
                img.setImageResource(imgRes[tabLayout.getSelectedTabPosition()]);
            }
        });
        img.startAnimation(fadeIn);
    }
}

File: BottomSheetSample/app/src/main/java/com/nickcode4fun/bottomsheetsample/MainActivity.kt
package com.nickcode4fun.bottomsheetsample

import android.os.Bundle
import android.support.design.widget.BottomSheetBehavior
import android.support.design.widget.BottomSheetDialog
import android.support.v7.app.AppCompatActivity
import android.view.View

class MainActivity : AppCompatActivity() {

    lateinit var bottomBehavior: BottomSheetBehavior<View>
    lateinit var bottomSheet: View

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.layout)
        bottomSheet = findViewById<View>(R.id.bottom_sheet)
        bottomBehavior = BottomSheetBehavior.from(bottomSheet)
    }

    fun showBottomSheet(v: View) {
        setBottomViewVisible(bottomBehavior.state != BottomSheetBehavior.STATE_EXPANDED)
    }

    fun showBottomSheetDialog(v: View) {
        val view: View = layoutInflater.inflate(R.layout.bottom_sheet, null)
        val dialog: BottomSheetDialog = BottomSheetDialog(this)
        dialog.setContentView(view)
        dialog.show()
    }

    fun showBottomSheetDialogFragment(v: View) {
        val bottomSheetFragment = BottomSheetFragment()
        bottomSheetFragment.show(supportFragmentManager, bottomSheetFragment.tag)
    }

    private fun setBottomViewVisible(isShow: Boolean) {
        if (isShow)
            bottomBehavior.state = BottomSheetBehavior.STATE_EXPANDED
        else
            bottomBehavior.state = BottomSheetBehavior.STATE_COLLAPSED
    }
}
